docs/content/sips/00x-cloudevent-trigger.md

```diff
@@ -0,0 +1,183 @@
+by defining the top-level application trigger:
+
+```toml
+# spin.toml
+trigger = { type = "cloudevent" }
+```
+
+Then, when defining the component (in `spin.toml`), you can set the protocol binding for the component. For example:
+
+- an HTTP CloudEvents component:
+
+```toml
+[component.trigger]
+binding = "http"
+```
+
+- an Kafka CloudEvents component (optional):
+
+```toml
+[component.trigger]
+binding = "kafka"
+broker = ["localhost:9092", "localhost:9093"]
+topic = "mytopic"
+group = "mygroup"
+```
+
+- an AMQP CloudEvents component (optional WIP):
+
+```toml
+[component.trigger]
+binding = "amqp"
+broker = "localhost:5672"
+exchange = "myexchange"
+routing_key = "myroutingkey"
+```
+
+You can also set the sink address for the component that returns a CloudEvent. For example:
+
+```toml
+[component.trigger]
+binding = "http"
+sink = "http://localhost:8080/someresource"
+```
+
+Note that the sink address is only used when the component is invoked. The component will make a outbound HTTP request that includes the CloudEvents to the sink address.
+
+
+## The WebAssembly interface
+
+The CloudEvent trigger is built on top of the
+[WebAssembly component model](https://github.com/WebAssembly/component-model).
+The current interface is defined using the
+[WebAssembly Interface (WIT)](https://github.com/bytecodealliance/wit-bindgen/blob/main/WIT.md)
+format, and is a function that takes the event payload as its only parameter:
+
+```fsharp
+// wit/ephemeral/spin-ce.wit
+
+// The event payload.
+record event {
+    // The event type.
+    type: string,
+    // The event id.
+    id: string,
+    // The event source.
+    source: string,
+    // The event specversion.
+    specversion: string,
+    // The event data content type.
+    datacontenttype: string,
+    // The event data schema.
+    dataschema: string,
+    // The event subject.
+    subject: string,
+    // The event time.
+    time: option<string>,
+    // The event data.
+    data: option<string>
+}
+
+// The entry point for a CloudEvent handler.
+handle-cloudevent: function(event: event) -> expected<event, error>
+```
+
+
+This is the function that all CloudEvents components must implement, and which is
+used by the Spin Redis executor when instantiating and invoking the component.
+
+Notice that the function will return a `expected<event, error>` value. If the sink address is not set, the function will return `expected<event, error>` with `event` as the value. If the sink address is set, spin will make an outbound HTTP request to the sink address and return the response as the value.
+
+Due to the [known issue](https://github.com/bytecodealliance/wit-bindgen/issues/171) of the cannonical ABI representation cannot exceeds the number of parameters (16) in the wasmtime API, the proposed WIT format cannot be compiled. I am proposing an alternative WIT format for CloudEvents:
+
+```fsharp
+// wit/ephemeral/spin-ce.wit
+
+// The event payload.
+type event = string
```

Comment:
Why `string` rather than `list<u8>`?

Reply:
discussed offline. I am fine with either of them since both are supported in cloudevents-sdk-rust